connectors/services/attachments.py

```diff
@@ -0,0 +1,250 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+""" File streamer service.
+
+Provides:
+
+- `AttachmentService` -- watches a dir for files to upload to Elasticsearch
+- `FileDrop` -- used by the producer to drop files on disk
+
+"""
+# TODO
+#
+# - harden all requests (catch errors, retries)
+# - if a file keep on failing, put it in a failing dir
+# - add a max file size
+#
+import os
+import json
+import functools
+import re
+
+import aiohttp
+import aiofiles
+
+from connectors.logger import logger
+from connectors.utils import (
+    CancellableSleeps,
+    get_event_loop,
+    ConcurrentTasks,
+)
+from connectors.services.base import BaseService
+
+
+SUPPORTED_TEXT_FILETYPE = [".py", ".rst", ".rb", ".sh", ".md", ".txt"]
+
+# 250MB max disk size
+ONE_MEG = 1024 * 1024
+DEFAULT_MAX_DIR_SIZE = 250
+
+# Program to encode in base64 -- we could compile a SIMD-aware one
+# for an extra performance boost
+BASE64 = "base64"
```

Comment:
Should discover the base64 binary once on startup and then use the absolute path throughout the execution of the program to avoid PATH poisoning.